profiling/src/internal/upscaling.rs

```diff
@@ -85,7 +85,7 @@ impl UpscalingRules {
                 let (_, rules) = self
                     .rules
                     .get_index_mut(index)
-                    .expect("Already existing rules");
+                    .ok_or(anyhow::anyhow!("Already existing rules"))?;
```

Comment:
I think we could improve on the messaging here while we are at it. Just an idea:
```suggestion
                    .ok_or(anyhow::anyhow!("Expected upscaling rule to exist for index {index}"))?;
```